### ISSUE TITLE
eth_call ignores patches

### ISSUE THREAD

**Issue by dimalit** [skalenetwork/skaled#1851]
When we use block timestamps to detect patches activation, it works in transactions, but is ignored in eth_call.
Problem is relevant only to new patch architecture (in old architecture it will use latest patches even on old blocks).

Test:
```
// SPDX-License-Identifier: GPL-3.0

pragma solidity >=0.8.2;

contract Push0Test {
    function f() public pure returns (string memory){
        return "success";
    }
}
```
Build it for Shanghai fork, deploy on a chain with PushZeroPatch, call f().
Expected: "success"
Actual: empty output + "bad instruction" exception

**Comment by DmytroNazarenko** [comment]
skaled: 3.19.0-beta.0
type: historic node

**Comment by oleksandrSydorenkoJ** [comment]
skaled: 3.19.0-beta.1


```
2024-05-31 16:22:21.215975   http://localhost:35992 >>> {"id":1610870113,"jsonrpc":"2.0","method":"eth_call","params":[{"data":"0x26121ff0","from":"0x71cBE3fedE33905d4D1Bf2Bd51f9d4A62375e659","to":"0x8fe575F9c3c7caEE1C0b86d05c7FB412d115B089","value":"0x0"},"0x1fb8b"]}
2024-05-31 16:22:21.216246   http://localhost:35992 <<< {"id":1610870113,"jsonrpc":"2.0","result":"0x"}
```

**Comment by oleksandrSydorenkoJ** [comment]
need to re-check on archival node

**Comment by oleksandrSydorenkoJ** [comment]
skaled: 3.19.0-beta.1
EVM-fork; shanghai

```
2024-06-04 15:39:04.740737   http://localhost:56116 >>> {"id":176330066,"jsonrpc":"2.0","method":"eth_call","params":[{"data":"0x0dbe671f","from":"0x71cBE3fedE33905d4D1Bf2Bd51f9d4A62375e659","to":"0x2DE9D37f7075617230DA85d76262c42aB2C91953","value":"0x0"},"0x289c5"]}
2024-06-04 15:39:04.741138   http://localhost:56116 <<< {"id":176330066,"jsonrpc":"2.0","result":"0x000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000077375636365737300000000000000000000000000000000000000000000000000"}

```

**Comment by oleksandrSydorenkoJ** [comment]
related to https://github.com/skalenetwork/skaled/issues/1896
Dencun (Deneb Cancun) fork

```
2024-05-31 16:22:21.215975   http://localhost:35992 >>> {"id":1610870113,"jsonrpc":"2.0","method":"eth_call","params":[{"data":"0x26121ff0","from":"0x71cBE3fedE33905d4D1Bf2Bd51f9d4A62375e659","to":"0x8fe575F9c3c7caEE1C0b86d05c7FB412d115B089","value":"0x0"},"0x1fb8b"]}
2024-05-31 16:22:21.216246   http://localhost:35992 <<< {"id":1610870113,"jsonrpc":"2.0","result":"0x"}
```

**Comment by DmytroNazarenko** [comment]
Closed as #1916 has more detailed information
